content: port onboarding doc from google docs (#25)
rhobs/handbook: content/Products/RHOBS/onboarding-a-tenant-into-rhobs.md

# Onboarding a Tenant into Red Hat’s Observability Service

- **Authors:**
  - [`@squat`](https://github.com/squat)

> ## Overview

Author: Hm, the title is not clear enough?

> ## Overview

Reviewer: 'onboarding a tenant' makes me think this is a set of actions that I (rhobs engineer) need to take rather than the other way round `how-to-onboard-into-rhobs` would have been my choice.

> ## Overview

Reviewer 2: That is a good point. I think it's as easy as being exploit in first sentence below, will propose something.

> Teams that have identified a need for collecting their logs and/or metrics into a centralized service that offers querying and dashboarding may choose to send their data to Red Hat’s hosted Observatorium instance. This document details the steps required to become a new tenant.

Reviewer 2: ```suggestion
Teams that have identified a need for collecting their logs and/or metrics into a centralized service that offers querying and dashboarding may choose to send their data to [RHOBS](README.md). This document details the required steps the interested party has to perform to become a new tenant.
```

Author: I kind of disagree here, there's no ambiguity IMO. `This document details the steps required to become a new tenant.` Who else would `become a new tenant`? :thinking:

Tenants who have been onboarded to the platform should consider validating their service accounts following the instructions in the [Testing Service Accounts for New Observatorium Tenants (internal)](https://docs.google.com/document/u/1/d/1CJsdGDnduA35Me6i3atHGDJnXzMOyWVKoqaAvFkdIbU/edit) doc.

> Once tenants have been onboarded to the platform and have validated their credentials, they can configure their clients to access Observatorium, following the instructions in the [Configuring Clients for Observatorium](configuring-clients-for-rhobs.md) doc.

Reviewer 2: ```suggestion
Once tenants have been onboarded to the platform and have validated their credentials, they can configure their clients to access RHOBS, following the instructions in the [Configuring Clients for RHOBS](configuring-clients-for-rhobs.md) doc.
```

## 1. Collect Requirements

> The first step is to collect information about the team’s requirements. The following questions should be answered:

Reviewer: ```suggestion
The first step is to collect information about the team’s requirements, this information will be required when the request to onboard into RHOBS is raised in section 5 below. The following questions should be answered:
```

1. What resources need to be collected? Logs? Metrics? Both?
2. What volume of data will be collected? XGb of logs per day? Xk metrics time series per day?
3. What is the expected or forecasted growth in collected data? X% growth in data per month?
4. How many users will require read access? X team members? The entire OpenShift org?
5. How many clients will be emitting data? 1 Prometheus server?

> 6. How long should data be retained? 2 weeks?

Author: Not today, but it's a valuable datapoint to know about prioritizing this feature and also knowing if we can fulfill their needs at all

> 6. How long should data be retained? 2 weeks?

Reviewer: Nice idea :+1:

> 6. How long should data be retained? 2 weeks?

Reviewer 2: I like the 6 questions with strong suggestion of the main option we have now ;p

7. How many independent service accounts do you need to write data into the platform?

> ## 2. Create a Customer Portal Account

Author: This is required and gets referenced a lot of times later on. Unfortunately this is the only way that red hat currently has for providing authz, i.e. via AMS :/ we really want to change this but can't today

> ## 2. Create a Customer Portal Account

Reviewer: Makes sense - worth adding that to the text then too.

> ## 2. Create a Customer Portal Account

Author: hmm I don't think that the new tenant needs to know the politics about why we have to use AMS but I agree they could benefit from more context about the need for this step

> ## 2. Create a Customer Portal Account

Reviewer 2: I would focus rather on creating Epic/Ticket which outlines today's pain points so we can figure out in the future.

> Create a Red Hat customer portal account to group all users from your team who should have access to the data. Note: any member of your account will be able to read the data collected by Observatorium for your tenant. https://sso.redhat.com

Reviewer 2: ```suggestion
Create a Red Hat customer portal account to group all users from your team who should have access to the data. Note: Any member of your account will be able to read the data collected by RHOBS for your tenant. 

Website: https://sso.redhat.com
```

## 3. Create Service Accounts for Reading/Writing Data

> All requests reading and writing data into the Observatorium platform must be authenticated and authorized. For this reason, each tenant must provision service accounts for their data writers, e.g. Prometheus, Promtail, etc, and their data readers, e.g. Grafana to be able to make requests. Services that would also like to test their configuration in a staging environment should create additional service accounts for the staging Observatorium instance. Services must have at least one service account, which they may choose to reuse among all of their data clients. If services would like to authenticate each client independently, then an additional service account is needed for each client. These service accounts, both for production and staging, must be created in the production external SSO environment, that is, sso.redhat.com. Service accounts may be requested via a ticket for the IT User team in Service Now at the URL help.redhat.com. The service account request should specify the following requirements:

Reviewer 2: ```suggestion
All requests reading and writing data into the RHOBS platform must be authenticated and authorized. For this reason, each tenant must provision service accounts for their data writers, e.g. Prometheus, Promtail, etc, and their data readers, e.g. Grafana to be able to make requests. Services that would also like to test their configuration in a staging environment should create additional service accounts for the staging Observatorium instance. Services must have at least one service account, which they may choose to reuse among all of their data clients. If services would like to authenticate each client independently, then an additional service account is needed for each client. These service accounts, both for production and staging, must be created in the production external SSO environment, that is, sso.redhat.com. Service accounts may be requested via a ticket for the IT User team in Service Now at the URL help.redhat.com. The service account request should specify the following requirements:
```

Reviewer 2: and all the mention of Observatorium really in this doc 😢

1. define two new service accounts in the production sso.redhat.com: observatorium-<tenant>, observatorium-<tenant>-staging;
2. enable the clients to use the client credentials grant flow;
3. add observatorium-telemeter to the allowed audience field of the observatorium-<tenant> account;
4. add observatorium-telemeter-staging to the allowed audience field of the observatorium-<tenant>-staging account;
5. set the email addresses of the accounts to: <your-email>@redhat.com;
6. set the preferred_username of the observatorium-<tenant> account to: service-account-observatorium-<tenant>; and
7. set the preferred_username of the observatorium-<tenant>-staging account to: service-account-observatorium-<tenant>-staging.
8. append one new allowed callback URL to the list of allowed callback URLs for the observatorium-telemeter service account: https://observatorium.api.openshift.com/oidc/<tenant>/callback
9. append one new allowed callback URL to the list of allowed callback URLs for the observatorium-telemeter-staging service account: https://observatorium.api.stage.openshift.com/oidc/<tenant>/callback

> ## 4. Authorize Service Accounts

Reviewer 2: I think what I am missing here is any mention about the infra tenant needs to send data to RHOBS, added ticket for this https://github.com/rhobs/handbook/issues/26
